perceptron.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Sep 30 16:05:24 2022
+
+@author: Ted
+"""
+
+# TODO: (M.Kouremetis) I think we should use numpy matrix operations versus iterating over lists

```

Comment:
I was thinking the same thing, just didn't have a chance to do it because it's a lot of changes.